igvm/templates/aws_user_data.cfg

```diff
@@ -0,0 +1,37 @@
+#cloud-config
+
+cloud_config_modules:
+  - apt-configure
+  - runcmd
+
+manage_etc_hosts: localhost
+hostname: {{ hostname }}
+fqdn: {{ fqdn }}
+
+apt:
+  sources:
+    {{ innogames_stable_name }}:
+      source: "{{ innogames_stable_source }}"
+      key: |
+        {{ innogames_stable_key }}
+    {{ basestretch_stable_name }}:
+      source: "{{ basestretch_stable_source }}"
+      key: |
+        {{ basestretch_stable_key }}
+
+package_update: true
+package_upgrade: true
+packages:
+ - puppet-agent
+ - puppet-msgpack
+
+runcmd:
+  - echo "# We need to use external IPs for puppet here" >> /etc/hosts
+  - echo {{ puppet_ca_addr }} {{ puppet_ca }} >> /etc/hosts
+  - echo {{ puppet_master_addr }} {{ puppet_master }} >> /etc/hosts
```

Comment:
The use of /etc/hosts was always strongly discouraged, if you'd use dns names, you wouldn't have to do this.

Reply:
@Woellchen Can you shed some light here again? Or isn't this needed anymore?

Reply:
in case of external dcs and puppet this is still necessary and won't change in near future because of different reasons i don't really feel like exposing on a public github repo :)

Reply:
I've added a TODO